go/obscuronode/host/host.go

```diff
@@ -329,32 +335,38 @@ func (a *Node) processBlocks(blocks []obscurocommon.EncodedBlock, interrupt *int
 
 	if !result.IngestedBlock {
 		b := blocks[len(blocks)-1].DecodeBlock()
-		log.Log(fmt.Sprintf(">   Agg%d: Did not ingest block b_%d", obscurocommon.ShortAddress(a.ID), obscurocommon.ShortHash(b.Hash())))
+		log.Log(fmt.Sprintf(">   Agg%d: Did not ingest block b_%d. Cause: %s", obscurocommon.ShortAddress(a.ID), obscurocommon.ShortHash(b.Hash()), result.BlockNotIngestedCause))
 		return
 	}
 
 	// Nodes can start before the genesis was published, and it makes no sense to enter the protocol.
 	if result.ProducedRollup.Header != nil {
 		a.P2p.BroadcastRollup(nodecommon.EncodeRollup(result.ProducedRollup.ToRollup()))
 
-		obscurocommon.ScheduleInterrupt(a.cfg.GossipRoundDuration, interrupt, func() {
-			if atomic.LoadInt32(a.interrupt) == 1 {
-				return
-			}
-			// Request the round winner for the current head
-			winnerRollup, submit := a.Enclave.RoundWinner(result.L2Hash)
-			if submit {
-				txData := obscurocommon.L1TxData{TxType: obscurocommon.RollupTx, Rollup: nodecommon.EncodeRollup(winnerRollup.ToRollup())}
-				tx := obscurocommon.NewL1Tx(txData)
-				t, err := obscurocommon.EncodeTx(tx)
-				if err != nil {
-					panic(err)
-				}
-				a.ethereumNode.BroadcastTx(t)
-				// collect Stats
-				// a.stats.NewRollup(DecodeRollupOrPanic(winnerRollup))
+		obscurocommon.ScheduleInterrupt(a.cfg.GossipRoundDuration, interrupt, a.handleHeader(result))
+	}
+}
+
+func (a *Node) handleHeader(result nodecommon.BlockSubmissionResponse) func() {
+	return func() {
+		if atomic.LoadInt32(a.interrupt) == 1 {
+			return
+		}
+		// Request the round winner for the current head
+		winnerRollup, winner, err := a.Enclave.RoundWinner(result.L2Hash)
```

Comment:
maybe an even better name is: "isWinner"

Reply:
Will fix.